WalletWasabi/WabiSabi/Models/DecompositionAlgs/Coin.cs

```diff
@@ -0,0 +1,40 @@
+using NBitcoin;
+using System;
+
+namespace WalletWasabi.WabiSabi.Models.DecompositionAlgs
+{
+	public record Coin : IComparable
+	{
+		public Money Amount { get; init; } = Money.Zero;
+		public const int OutputVbytes = 31;
+		public const decimal InputVbytes = 68.5m;
+
+		private int HashCode { get; } = new Guid().GetHashCode();
+
+		public Money EffectiveValue(FeeRate feeRate)
+		{
+			long sats = (long)Math.Floor(feeRate.SatoshiPerByte * InputVbytes);
+			return Amount - Money.Satoshis(sats);
+		}
+
+		public Money EffectiveCost(FeeRate feeRate)
+		{
+			long sats = feeRate.GetFee(OutputVbytes);
+			return Amount + Money.Satoshis(sats);
```

Comment:
??? I don't get it. If the coin has a value of, let say, 100sats and the effective value is 85sats (15sats are needed for fees) then the cost is 15sats or we could say that the for having an output of 85sats it will cost us 100sats but why is this effective cost defined as the amount + the fees?

Reply:
because the balance proof etc need the effective cost, it's just the total cost of the output registration not the nominal value of the resulting output.